src/libsync/syncjournaldb.cpp

```diff
@@ -1245,6 +1259,55 @@ void SyncJournalDb::setPollInfo(const SyncJournalDb::PollInfo& info)
     }
 }
 
+QStringList SyncJournalDb::getSelectiveSyncList(SyncJournalDb::SelectiveSyncListType type)
+{
+    QStringList result;
+
+    QMutexLocker locker(&_mutex);
+    if( !checkConnect() ) {
+        return result;
+    }
+
+    _getSelectiveSyncListQuery->reset();
+    _getSelectiveSyncListQuery->bindValue(1, int(type));
+    if (!_getSelectiveSyncListQuery->exec()) {
+        qWarning() << "SQL query failed: "<< _getSelectiveSyncListQuery->error();
+        return result;
+    }
+    while( _getSelectiveSyncListQuery->next() ) {
+        auto entry = _getSelectiveSyncListQuery->stringValue(0);
+        if (!entry.endsWith(QLatin1Char('/'))) {
+            entry.append(QLatin1Char('/'));
+        }
+        result.append(entry);
+    }
+    return result;
+}
+
+void SyncJournalDb::setSelectiveSyncList(SyncJournalDb::SelectiveSyncListType type, const QStringList& list)
+{
+    QMutexLocker locker(&_mutex);
+    if( !checkConnect() ) {
+        return;
+    }
+
+    //first, delete all entries of this type
+    SqlQuery delQuery("DELETE FROM selectivesync WHERE type == ?1", _db);
+    delQuery.bindValue(1, int(type));
+    if( !delQuery.exec() ) {
+        qWarning() << "SQL error when deleting selective sync list" << list << delQuery.error();
+    }
+
+    SqlQuery insQuery("INSERT INTO selectivesync VALUES (?1, ?2)" , _db);
```

Comment:
not all statements are prepared. in general only the statements that happens often during the sync are.  But maybe it deserve to be prepared.